File: account/src/main/java/serp/project/account/core/domain/dto/request/UpgradeSubscriptionRequest.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.account.core.domain.dto.request;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import serp.project.account.core.domain.enums.BillingCycle;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class UpgradeSubscriptionRequest {

    @NotNull(message = "New plan ID is required")
    private Long newPlanId;

    private BillingCycle billingCycle;

    @Builder.Default
    private Boolean isAutoRenew = false;

    private String notes;
}

File: api_gateway/src/core/domain/dto/request/account/upgrade_subscription_request.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package request

type UpgradeSubscriptionRequest struct {
	NewPlanId    int64   `json:"newPlanId"`
	BillingCycle string  `json:"billingCycle"`
	IsAutoRenew  *bool   `json:"isAutoRenew"`
	Notes        *string `json:"notes"`
}
